File: shoring/blackjackEngine/src/main/java/com/panopset/blackjackEngine/DeckPile.java
package com.panopset.blackjackEngine;

import com.panopset.compat.FlagSwitch;

public enum DeckPile {

  INSTANCE;

  private FlagSwitch flag = new FlagSwitch();

  
  public static void reset() {
    INSTANCE.flag.reset();
  }

  public static boolean pull() {
    return INSTANCE.flag.pull();
  }
}

File: code/shoring/blackjackEngine/src/main/kotlin/com/panopset/blackjackEngine/DeckPile.kt
package com.panopset.blackjackEngine

import com.panopset.compat.FlagSwitch

object DeckPile {

    private val flag = FlagSwitch()

    fun reset() {
        flag.reset()
    }

    fun pull(): Boolean {
        return flag.pull()
    }
}
